### PR TITLE
feat(glue-alpha): adding partition projection

### PR DESCRIPTION
**Pull request by guyernest**
The [partition projection feature in Athena and Glue](https://docs.aws.amazon.com/athena/latest/ug/partition-projection.html) needed to be supported correctly in the S3Table implementation. The table properties from the partition project were not added to the Parameters section of TableInput.

The new implementation allows the definition of the different types of partition projection, including date, integer and enum,

Closes https://github.com/aws/aws-cdk/issues/27365

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-glue-alpha/test/s3-table.test.ts [hunk 1]
```diff
@@ -1095,6 +1096,145 @@ describe('validate', () => {
...
+    const partitionProjection = new DatePartitionProjection(
+        'columnName',
+        's3://DOC-EXAMPLE-BUCKET/prefix/${columnName}/',
+        'NOW-3YEARS,NOW',
+        'dd-MM-yyyy',
+        1,
+        DateIntervalUnit.DAYS,
+    );
+  
+    const table = new glue.S3Table(stack, 'Table', {
```
Comment (scanlonp) [inline packages/@aws-cdk/aws-glue-alpha/test/s3-table.test.ts:1112]:
```suggestion
    new glue.S3Table(stack, 'Table', {
```

File: packages/@aws-cdk/aws-glue-alpha/test/s3-table.test.ts [hunk 2]
```diff
@@ -1095,6 +1096,145 @@ describe('validate', () => {
...
+    const partitionProjection = new DatePartitionProjection(
+        'columnName',
+        's3://DOC-EXAMPLE-BUCKET/prefix/${columnName}/',
+        '2021/01/01,NOW',
+        'yyyy/MM/dd',
+        1,
+        DateIntervalUnit.DAYS,
+    );
+  
+    const table = new glue.S3Table(stack, 'Table', {
```
Comment (scanlonp) [inline packages/@aws-cdk/aws-glue-alpha/test/s3-table.test.ts:1150]:
```suggestion
    new glue.S3Table(stack, 'Table', {
```

**Review by scanlonp — CHANGES_REQUESTED** [review]
Thank you for the contribution! A couple comments.

First, we require that pulls are made from non-main branches on your fork (see https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#step-4-pull-request for more info). Please move this on to a different branch.

Removing the unused variable names in the tests should fix the immediate errors in the build. Then we can give a meaningful review.

**Comment by guyernest** [comment]
Cleaning and opening a new PR

**Pull request closed without merging**